config.nix

```diff
@@ -349,7 +349,14 @@ let
     toplevelCargotoml = readTOML (root + "/Cargo.toml");
 
     # The cargo lock
-    cargolock = readTOML (root + "/Cargo.lock");
+    cargolock = 
+      let
+        cargolock-file = root + "/Cargo.lock";
+      in
+      if builtins.pathExists cargolock-file then
```

Comment:
Did you try this with IFD, i.e. when the source is a built derivation in the store?

Reply:
Not if it is not in the CI.